session-5/task2.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import "fmt"
+
+func incrementbyValue(a int) int {
+	a = a + 1
+	return a
+}
+
+func incrementbyPointer(x *int) {
+	*x = *x + 1
+	// Question
+	//why we can't use  ??
+	//a =*x +1
+	//return a
```

Comment:
Effektiv kod en vacib shertlerden biridir, ona gore muveqqeti deyer yaradib orda hesablama edib yeniden onu oturmekle cost yaradiriq
```suggestion
	*x += 1
```